packages/web/src/shared/components/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,123 @@
+import { Dialog, Transition } from '@headlessui/react';
+import {
+  Fragment,
+  forwardRef,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import Button from './actions/Button';
+import TextInput from './form/TextInput';
+
+export interface ConfirmationDialogProps {
+  title: string;
+  description: string;
+  confirmationValue: string;
+}
+
+export interface ConfirmationDialogRef {
+  open(): Promise<boolean>;
+}
+
+const ConfirmationDialog = forwardRef<
+  ConfirmationDialogRef,
+  ConfirmationDialogProps
+>(({ title, description, confirmationValue }: ConfirmationDialogProps, ref) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      async open() {
+        return new Promise<boolean>((resolve) => {
+          setIsOpen(true);
+          // TODO: how do I call resolve when the confirm button is pressed, or the dialog is closed?
+
+          // setIsOpen(false);
+          // resolve(true);
+        });
+      },
```

Comment:
@arrocke I got stuck on how to implement this.

Reply:
You can set resolve to a ref in this function. Then the click/submit handler for the modal can call the function from the ref

Reply:
I got it working. I'd appreciate any advice on cleaning up the code.